benchmark_controller.py

```diff
@@ -139,19 +139,27 @@ def _load_models(self):
         self._build_complete_flags()
 
 
-    def _post_run(self, stdout, perf_results):
+    def _post_run(self, stdouts, perf_results, metadatas):
         """This function executes after the benchmark has been run"""
-        self.logger.debug(stdout)
+        self.logger.debug(stdouts)
         self.logger.debug(perf_results)
-        self._output_logs(stdout, perf_results)
+        self.logger.debug(metadatas)
+        for i in range(0, len(metadatas)):
+            self._output_logs(stdouts[i], perf_results[i], metadatas[i])
```

Comment:
implicit assumption that all three will have the same size. Maybe this needs a structure to contain all three results and them build a list of that?

Reply:
They are all the same size that's a constraint put by the run_all(run_benchmark())